Tests/FileIO/TestXmlGmlReader.cpp

```diff
@@ -19,120 +19,242 @@
+			EXPECT_EQ(ids[2], tri[2]);
+		};
+
+		auto checkSurface = [&checkTriangleIDs](GeoLib::SurfaceVec const& sfcs,
+			std::size_t sfc_id, std::vector<std::array<std::size_t,3>> tri_ids,
+			std::string const& name)
+		{
+			auto const& sfc_vec = *(sfcs.getVector());
+			auto const& sfc = *(sfc_vec[sfc_id]);
+			EXPECT_EQ(tri_ids.size(), sfc.getNTriangles());
+			for (std::size_t k(0); k<tri_ids.size(); ++k)
+				checkTriangleIDs(*(sfc[k]), tri_ids[k]);
+
+			std::string sfc_name;
+			sfcs.getNameOfElementByID(sfc_id, sfc_name);
+			EXPECT_EQ(0u, name.compare(sfc_name));
+		};
+
+		auto const read_sfcs = geo_objects.getSurfaceVecObj(geo_name);
+		EXPECT_EQ(2u, read_sfcs->size());
+		checkSurface(*read_sfcs, 0,
+			{{{0,3,1}}, {{1,3,4}}, {{1,4,2}}, {{2,4,5}}}, "");
+		checkSurface(*read_sfcs, 1, {{{3,6,8}}, {{3,8,5}}}, "SecondSurface");
+	}
+};
+
+TEST_F(FileIOXmlGml, QtXmlGmlWriterReaderTest)
+{
+	// Writer test
+	std::string test_data_file(BaseLib::BuildInfo::tests_tmp_path + "xmlgmltestdata.gml");
+
 	FileIO::XmlGmlInterface xml(geo_objects);
 	xml.setNameForExport(geo_name);
 	int result = xml.writeToFile(test_data_file);
-	ASSERT_EQ(result, 1);
+	EXPECT_EQ(result, 1);
+
+	// remove the written data from the data structures
+	geo_objects.removeSurfaceVec(geo_name);
+	geo_objects.removePolylineVec(geo_name);
+	geo_objects.removePointVec(geo_name);
 
 	// Reader test
 	result = xml.readFile(QString::fromStdString(test_data_file));
-	ASSERT_EQ(result, 1);
-
-	const std::vector<GeoLib::Point*> *readerPoints = geo_objects.getPointVec(geo_name);
-	const GeoLib::PolylineVec *line_vec = geo_objects.getPolylineVecObj(geo_name);
-	const std::vector<GeoLib::Polyline*> *readerLines = geo_objects.getPolylineVec(geo_name);
-	const std::vector<GeoLib::Surface*> *readerSfcs = geo_objects.getSurfaceVec(geo_name);
-	ASSERT_EQ(9u, readerPoints->size());
-	ASSERT_EQ(5u, readerLines->size());
-	ASSERT_EQ(2u, readerSfcs->size());
-
-	GeoLib::Point* pnt = (*readerPoints)[7];
-	ASSERT_EQ(3.0, (*pnt)[0]);
-	ASSERT_EQ(2.0, (*pnt)[1]);
-	ASSERT_EQ(0.0, (*pnt)[2]);
-
-	GeoLib::Polyline* line = (*readerLines)[4];
-	ASSERT_EQ(3u, line->getNumberOfPoints());
-	ASSERT_EQ(6u, line->getPointID(0));
-	ASSERT_EQ(7u, line->getPointID(1));
-	ASSERT_EQ(8u, line->getPointID(2));
-	std::string line_name("");
-	line_vec->getNameOfElementByID(4, line_name);
-	ASSERT_EQ("right", line_name);
-
-	GeoLib::Surface* sfc = (*readerSfcs)[1];
-	ASSERT_EQ(2u, sfc->getNTriangles());
-	const GeoLib::Triangle* tri = (*sfc)[1];
-	ASSERT_EQ(3u, (*tri)[0]);
-	ASSERT_EQ(8u, (*tri)[1]);
-	ASSERT_EQ(5u, (*tri)[2]);
+	EXPECT_EQ(1, result);
+
+	checkPointProperties();
+	checkPolylineProperties();
+	checkSurfaceProperties();
+
+	boost::filesystem::remove(test_data_file);
+	test_data_file += ".md5";
+	boost::filesystem::remove(test_data_file);
+}
+
+TEST_F(FileIOXmlGml, BoostXmlGmlWriterReaderTest)
+{
+	// Writer test
+	std::string test_data_file(BaseLib::BuildInfo::tests_tmp_path + "xmlgmltestdata.gml");
```

Comment:
Could be improved with boost's [`filesystem::unique_path()`](http://www.boost.org/doc/libs/master/libs/filesystem/doc/reference.html#unique_path) to avoid race-conditions on the filesystem or leftovers from last (e.g. segfaulted) run.

It, unfortunately, does not solve the problem of a dangling file like posix' `mkstemp`, which would remove the temporary after the program exits, or stream is closed; _but_ it is not portable.... :(